Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm

```diff
@@ -261,8 +261,26 @@ - (BOOL)prefersStatusBarHidden
     if (m_mockVideoPresentationModeEnabled)
         return;
 
-    if (RefPtr interface = findInterface(contextId))
+    if (RefPtr interface = findInterface(contextId)) {
         interface->documentVisibilityChanged(isDocumentVisible);
+        Ref model = ensureModel(contextId);
+        if (model->isChildOfElementFullscreen())
+            videosInElementFullscreenChanged();
```

Comment:
Why are we updating `videosInElementFullscreenChanged()` when the document becomes visible or not?

Reply:
I understood this method as being the video element becoming visible or not. As such it impacts the calculation of the most visible video element.